test/unit-test/twcc_manager/twcc_manager_utest.c

```diff
@@ -176,22 +462,223 @@ void test_twccFindPacket_Found( void )
         twccPacketInfo.packetSeqNum += seqNum;
         result = RtcpTwccManager_AddPacketInfo( &( twccManager ),
                                                 &( twccPacketInfo ) );
-        TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_OK, result );
+        TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_OK,
+                           result );
     }
 
-    TEST_ASSERT_EQUAL( TWCC_PACKET_INFO_ARRAY_LENGTH, twccManager.count );
+    TEST_ASSERT_EQUAL( TWCC_PACKET_INFO_ARRAY_LENGTH,
+                       twccManager.count );
 
     for( i = 0; i < TWCC_PACKET_INFO_ARRAY_LENGTH; i++ )
     {
         result = RtcpTwccManager_FindPacketInfo( &( twccManager ),
-                                                seqNum * ( i + 1 ),
-                                                &( twccPacketInfo ) );
+                                                 seqNum * ( i + 1 ),
+                                                 &( twccPacketInfo ) );
+
+        TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_OK,
+                           result );
+        TEST_ASSERT_EQUAL( i,
+                           twccPacketInfo.packetSize );
+        TEST_ASSERT_EQUAL( seqNum * ( i + 1 ),
+                           twccPacketInfo.packetSeqNum );
+        TEST_ASSERT_EQUAL( localSentTime + ( int64_t ) i,
+                           twccPacketInfo.localSentTime );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate Twcc Manager Handle packet fail functionality for Bad Parameter.
+ */
+void test_twccHandlePacket_BadParams( void )
+{
+    RtcpTwccManager_t twccManager;
+    RtcpTwccManagerResult_t result;
+    RtcpTwccPacket_t twccPacket;
+    TwccBandwidthInfo_t twccBandwidthInfo = { 0 };
+
+    result = RtcpTwccManager_HandleTwccPacket( NULL,
+                                               &( twccPacket ),
+                                               &( twccBandwidthInfo ) );
+    TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_BAD_PARAM,
+                       result );
+
+    result = RtcpTwccManager_HandleTwccPacket( &( twccManager ),
+                                               NULL,
+                                               &( twccBandwidthInfo ) );
+    TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_BAD_PARAM,
+                       result );
+
+    result = RtcpTwccManager_HandleTwccPacket( &( twccManager ),
+                                               &( twccPacket ),
+                                               NULL );
+    TEST_ASSERT_EQUAL( RTCP_TWCC_MANAGER_RESULT_BAD_PARAM,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate Twcc Manager Handle packet functionality for No Packets.
```

Comment:
What does no packets mean here? And why does RtcpTwccManager_HandleTwccPacket() returns OK when no packets?

Reply:
I think its because since no packet is being added in that test case, The twccHandler runs smoothly and doesn't modify the bandwidth Information. In the flow I verified for finding the packet we are getting `RTCP_TWCC_MANAGER_RESULT_EMPTY`, but the handler is just supposed to change the bandwidth info, since no packet came. It won't do it and exits with `RTCP_TWCC_MANAGER_RESULT_OK`.

Reply:
One following question for this, what does it return in the twccBandwidthInfo? Is it ok for user to respond with this bandwidth info?

Reply:
So the TWCCHandle takes the already existing bandwidth info ( for the unit tests we have initialised all its parameters to 0, but in real scenario there would an existing bandwidth info that the twcc handle has used earlier.) Since No packets are arriving , the handle doesn't modify the bandwidth info and returns the bandwidth info it took as it is. 

It should be ok for the user to respond with this bandwidth info cause this is the bandwidth utilised so far, since this last operation didn't modify it cause of no packets. It should be fine returning OK.